vitrina/datasets/models.py

```diff
@@ -1383,8 +1381,8 @@ def update_applicable_legislation(self, urls: list[str]) -> None:
         existing_urls = set(ApplicableLegislation.objects.filter(url__in=urls).values_list("url", flat=True))
         new_urls = [url for url in urls if url not in existing_urls]
 
-        if new_urls:
-            ApplicableLegislation.objects.bulk_create([ApplicableLegislation(url=url) for url in new_urls])
+        for url in new_urls:
```

Comment:
You're moving from `bulk_create` due to reversion, but this is not enough then:
1) Let's collect the records being created, to not have to call the database to get all of them
2) Have you checked for timeouts? How many URL's do I need to add to the form, to crash the web with a timeout? Have you considered celery for this?